File: 2020-Spring/COMSC-275 Intro to Web Programming/Assignments/7. StatCalcObj/Python/statcalc.py
class StatCalc:
    def __init__(self,dataList):
        self.dataList=dataList
        self.sdataList=[]+ self.dataList
        self.sdataList.sort()

    def showResults(self):
        print("Original Values: <br>")
        for item in self.dataList:
            print(item)
            print(" ")

        print("<br><br>")
        print("Sorted Values: <br>")
        for item in self.sdataList:
            print(item)
        print("<br>")

    def findMin(self):
        return self.sdataList[0]

    def findMax(self):
        return self.sdataList \
            [len(self.sdataList)-1]

    def findMean(self):
        mean = sum(self.sdataList) / \
               len(self.sdataList)
        return mean

    def findMedian(self):
        mid = len(self.sdataList) / 2
        if len(self.sdataList) % 2 == 0:  # if even set
            result = (self.sdataList[int(mid)] + self.sdataList[int(mid - 1)]) / 2
        else:
            result = self.sdataList[int(mid - 0.5)]
        return result

    def calcValues(self,opList):
        dictOpRes = {}
        for op in opList:
            if op == "min":
                dictOpRes[op] = self.findMin()
            if op == "max":
                dictOpRes[op] = self.findMax()
            if op == "mean":
                dictOpRes[op] = self.findMean()
            if op == "median":
                dictOpRes[op] = self.findMedian()

        return dictOpRes

File: 2020-Spring/COMSC-275 Intro to Web Programming/Assignments/7. StatCalcObj/PHP/StatCalc.php
<?php
class StatCalc {
    
    private $data;
    
    private $sdata;
    
    public function __construct ($d){
        
        
        
        $this->data = $d;
        
        $this->sdata = $this->data;
        
        sort ($this->sdata);
        
    }
    
    public function findMin(){
        
        return $this->sdata[0];
        
    }
    
    public function findMax (){
        
        return $this->sdata[count($this->sdata)-1];
        
    }
    
    public function findMean (){
        
        $sum = array_sum($this->sdata);
        
        $mean = $sum / count ($this->sdata);
        
        return $mean;
        
    }
    
    public function findMedian (){
        
        //Determine if the length is odd or even.
        
        if ( (count($this->sdata) % 2) != 0 ){
            
            $index = count ($this->sdata) / 2;
            
            $median = $this->sdata [$index];
            
        }
        
        else{
            
            $indexHi = count ($this->sdata) / 2;           
            $indexLo = $indexHi - 1;
            
            $median = ($this->sdata[$indexLo] +
                
                $this->sdata [$indexHi]) / 2;
                
        }
        
        return $median;        
    }
    
    public function getOrigData(){
        
        return $this->data;        
    }
    
    public function getSdataData(){
        
        return $this->sdata; 
    }
    
}

?>